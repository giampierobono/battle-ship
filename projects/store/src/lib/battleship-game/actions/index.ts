export * from './battle-ship-game.actions';
