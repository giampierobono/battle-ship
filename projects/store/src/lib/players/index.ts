export * from './models';
export * from './players.actions';
export * from './players.reducer';
