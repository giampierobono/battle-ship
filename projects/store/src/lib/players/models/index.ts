export * from './add-boat-to-player-payload.model';
export * from './create-player-payload.model';
export * from './hit-player-boat-payload.model';
