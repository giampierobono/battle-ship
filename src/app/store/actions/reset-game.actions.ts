import { createAction } from '@ngrx/store';

export const resetGame = createAction('[BS webapp] Reset game');
