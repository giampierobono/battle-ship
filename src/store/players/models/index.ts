export * from './add-players-payload.model';
export * from './create-player-payload.model';
