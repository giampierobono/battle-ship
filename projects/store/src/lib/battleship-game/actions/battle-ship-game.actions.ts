import { createAction, props } from '@ngrx/store';
import { PlayerFiresPositionPayloadModel } from '../../players';

export const tryFirePosition = createAction('[Game] Player fire!', props<PlayerFiresPositionPayloadModel>());
