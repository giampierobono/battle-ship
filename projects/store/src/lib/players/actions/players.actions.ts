import { createAction, props } from '@ngrx/store';
import { AddBoatToPlayerPayload, CreatePlayerPayloadModel, HitPlayerBoatPayloadModel } from '../models';

export const createPlayer = createAction('[Players] Add new player', props<CreatePlayerPayloadModel>());

export const addBoatToPlayer = createAction('[Players] Add boat to player', props<AddBoatToPlayerPayload>());

export const hitPlayerBoat = createAction('[Player] Hit player boat', props<HitPlayerBoatPayloadModel>());
