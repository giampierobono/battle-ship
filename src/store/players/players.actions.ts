import { createAction, props } from '@ngrx/store';
import { AddPlayersPayloadModel } from './models';

export const addPlayers = createAction('[Players] Add new players', props<AddPlayersPayloadModel>());
