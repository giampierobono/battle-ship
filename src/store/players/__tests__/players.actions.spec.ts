import { addPlayers } from '../players.actions';

describe('Players action', () => {
  describe('addPlayers', () => {
    it('should return correct action structure', () => {
      expect(addPlayers({ players: [] })).toMatchSnapshot();
    });
  });
});
