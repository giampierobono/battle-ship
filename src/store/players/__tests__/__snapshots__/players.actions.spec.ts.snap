// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Players action addPlayers should return correct action structure 1`] = `
Object {
  "players": Array [],
  "type": "[Players] Add new players",
}
`;
